charts/datadog/templates/_container-trace-agent.yaml

```diff
@@ -2,12 +2,32 @@
 - name: trace-agent
   image: "{{ include "image-path" (dict "root" .Values "image" .Values.agents.image) }}"
   imagePullPolicy: {{ .Values.agents.image.pullPolicy }}
+  {{- $version := .Values.agents.image.tag | toString | trimSuffix "-jmx" -}}
+  {{- $length := len (split "." $version) -}}
+  {{- if and (eq $length 1) (eq $version "6") -}}
+  {{- $version = "6.48.0" -}}
```

Comment:
We could maybe default to `[6,7].46.0` for now (latest stable at the time of this writing), and update this once `7.48.x` is available in a few weeks. That would allow us to merge this immediately.

Reply:
what about custom versions?
Can we have a config flag to override the check here?

Reply:
The concern - debug version or special version we provide to customers, or use internally

but that leads me into a realization that breaking the command line is problematic and can introduce friction with customers
Shouldn't we allow backward compatibility for a couple of version?